ui/app/src/components/EditModeSwitch/EditModeSwitch.tsx

```diff
@@ -100,7 +101,9 @@ export function EditModeSwitch(props: EditModeSwitchProps) {
   const editMode = useSelection((state) => state.preview.editMode) && !isLocked && write;
 
   const onChange = (e) => {
-    dispatch(setPreviewEditMode({ editMode: e.target.checked }));
+    dispatch(
+      batchActions([conditionallyUnlockItem({ path: item.path }), setPreviewEditMode({ editMode: e.target.checked })])
+    );
```

Comment:
We should conditionally unlock on an epic based on preview mode so we don't have to duplicate code on all the switches that toggle editing.